code_samples/discounts/src/Command/ManageDiscountsCommand.php

```diff
@@ -0,0 +1,88 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Command;
+
+use Ibexa\Contracts\Core\Collection\ArrayMap;
+use Ibexa\Contracts\Core\Repository\PermissionResolver;
+use Ibexa\Contracts\Core\Repository\UserService;
+use Ibexa\Contracts\Discounts\DiscountServiceInterface;
+use Ibexa\Contracts\Discounts\Value\DiscountType;
+use Ibexa\Contracts\Discounts\Value\Struct\DiscountCreateStruct;
+use Ibexa\Contracts\Discounts\Value\Struct\DiscountTranslationStruct;
+use Ibexa\Contracts\DiscountsCodes\DiscountCodeServiceInterface;
+use Ibexa\Contracts\DiscountsCodes\Value\Struct\DiscountCodeCreateStruct;
+use Ibexa\Discounts\Value\DiscountCondition\IsInCurrency;
+use Ibexa\Discounts\Value\DiscountCondition\IsInRegions;
+use Ibexa\Discounts\Value\DiscountCondition\IsProductInArray;
+use Ibexa\Discounts\Value\DiscountRule\FixedAmount;
+use Ibexa\DiscountsCodes\Value\DiscountCondition\IsValidDiscountCode;
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+
+final class ManageDiscountsCommand extends Command
+{
+    protected static $defaultName = 'discounts:manage';
+
+    private DiscountServiceInterface $discountService;
+
+    private DiscountCodeServiceInterface $discountCodeService;
+
+    private PermissionResolver $permissionResolver;
+
+    private UserService $userService;
+
+    public function __construct(UserService $userSerice, PermissionResolver $permissionResolver, DiscountServiceInterface $discountService, DiscountCodeServiceInterface $discountCodeService)
+    {
+        $this->userService = $userSerice;
+        $this->discountService = $discountService;
+        $this->discountCodeService = $discountCodeService;
+        $this->permissionResolver = $permissionResolver;
+
+        parent::__construct();
+    }
+
+    protected function execute(InputInterface $input, OutputInterface $output): int
+    {
+        $this->permissionResolver->setCurrentUserReference($this->userService->loadUserByLogin('admin'));
+
+        $now = new \DateTimeImmutable();
+
+        $discountCodeCreateStruct = new DiscountCodeCreateStruct(
+            'summer10',
+            null, // Unlimited usage
+            $this->permissionResolver->getCurrentUserReference()->getUserId(),
+            $now
+        );
+        $discountCode = $this->discountCodeService->createDiscountCode($discountCodeCreateStruct);
+
+        $discountCreateStruct = new DiscountCreateStruct();
+        $discountCreateStruct->setIdentifier('discount_identifier')
```

Comment:
I've chosen the fluent API usage over the constructor, as this is self-documenting for the code sample - I'd just use the constructor in "standard" code. 

I hope it will be clear for the readers that they can just use the constructor.